From 6acecdeace5d508c7ecf391412560b3edb1baaec Mon Sep 17 00:00:00 2001
From: Zhou Zheng <zhouzheng2069@phytium.com.cn>
Date: Mon, 17 Jun 2024 19:33:07 +0800
Subject: [PATCH 083/150] ASoC: codec: Ensure to use appropriate level to
 enable the SPK

Speaker has two enabling ways(high or low level),
so it add operation of reading the device tree property.
Ensure to use appropriate level to enable the speaker.

Add a variable to acquire the spk-active-level and change
enabling relationship of hp_inserted.

Signed-off-by: Zhou Zheng <zhouzheng2069@phytium.com.cn>
Signed-off-by: Wang Yinfeng <wangyinfeng@phytium.com.cn>
Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Change-Id: Ieeecb530ad922c0251a8f51c88ca53a29d7f283b
Signed-off-by: Andrew Powers-Holmes <aholmes@omnom.net>
---
 sound/soc/codecs/es8336.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/sound/soc/codecs/es8336.c b/sound/soc/codecs/es8336.c
index 4e224b035c3b..acb055bfaab2 100644
--- a/sound/soc/codecs/es8336.c
+++ b/sound/soc/codecs/es8336.c
@@ -880,7 +880,7 @@ static void hp_work(struct work_struct *work)
 	if (es8336->hp_det_invert)
 		enable = !enable;
 
-	es8336->hp_inserted = enable ? true : false;
+	es8336->hp_inserted = !enable;
 	if (!es8336->muted) {
 		if (es8336->hp_inserted)
 			es8336_enable_spk(es8336, false);
@@ -964,6 +964,7 @@ static int es8336_i2c_probe(struct i2c_client *i2c)
 	struct gpio_desc *gpiod;
 	int ret = -1;
 	int hp_irq;
+	int active_level = 0;
 
 	es8336 = devm_kzalloc(&i2c->dev, sizeof(*es8336), GFP_KERNEL);
 	if (!es8336)
@@ -986,13 +987,13 @@ static int es8336_i2c_probe(struct i2c_client *i2c)
 
 	gpiod = devm_gpiod_get_index_optional(&i2c->dev, "sel", 0,
 							GPIOD_OUT_HIGH);
-
+	device_property_read_u32(&i2c->dev, "spk-active-level", &active_level);
 	if (!gpiod) {
 		dev_info(&i2c->dev, "Can not get spk_ctl_gpio\n");
 		es8336->spk_ctl_gpio = INVALID_GPIO;
 	} else {
 		es8336->spk_ctl_gpio = desc_to_gpio(gpiod);
-		es8336->spk_active_level = 0;
+		es8336->spk_active_level = active_level;
 		es8336_enable_spk(es8336, false);
 	}
 
-- 
2.47.0

